File: frontend/src/types/notification.ts
// ============================================================
// Company OS — Notification Types
// ============================================================

export interface Notification {
  id: string;
  user_id: string;
  type: string;
  title: string;
  message: string;
  entity_type?: string;
  entity_id?: string;
  read: boolean;
  created_at: string;
}

File: app/schemas/notification.py
import uuid
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class NotificationOut(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    type: str
    title: str
    message: str
    entity_type: Optional[str] = None
    entity_id: Optional[uuid.UUID] = None
    read: bool
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)
